### ISSUE TITLE
Documentation: Add documentation for different use cases and examples

### ISSUE THREAD

**Issue by junwen-k** [fumiama/go-docx#11]
A good open source library cannot go without a good documentation!

Suggest to add in documentation for different use cases and examples.
A good starting point could be based on the existing features:

1. Text
1. Image
1. Table
1. Shape
1. Canvas
1. Group

Maybe it can also be linked back to Microsoft OpenXML's documentation for easier references.

Easier way to get started is by directly writing into the repository's [README file](https://github.com/fumiama/go-docx/blob/master/README.md) but it could get cluttered real quick.

Perhaps having a Nextra documentation site would be helpful.
https://nextra.site/docs/docs-theme/start

**Comment by fumiama** [comment]
That's a good idea and we can use github wiki as the `extra documentation site`. I'll be going on it as soon as I'm free. Anyone who is interested in this project can also help us to improve the documentation by PR or opening issues.

**Comment by lordofscripts** [comment]
I agree that documentation is required, otherwise what's the use of it?

**Comment by saithan-ph** [comment]
am sorry. i try to Replace or Edit Text in Paragraphs like this or some way. 
"{{NAME}}": "John Doe",
"{{DATE}}": "October 15, 2024",

but can't find example.

Thank you.

**Comment by dcharles525** [comment]
I want to take this on, as I have largely documented a lot of the functions in my own project. @fumiama Any particular format you're looking for? I was going to start with a Markdown file with documentation and short examples.

**Comment by fumiama** [comment]
> I want to take this on, as I have largely documented a lot of the functions in my own project. @fumiama Any particular format you're looking for? I was going to start with a Markdown file with documentation and short examples. Then add it to the wiki. 

Thanks so much for your effort. Maybe we can start from adding the comments embedded in .go files. It supports some simple markdown grammars and will show the help of a certain function or variable in IDE when the mouse hovers.

As these are just some simple descriptions, for the sdvanced usage, we can open a wiki page or create some md files in `doc` folder. Of course some example codes are acceptable. It can be put in `example` folder.
